File: client/packages/domain/lib/src/room/repositories/browse_room_repository.dart
import 'package:core/errors.dart';
import 'package:fpdart/fpdart.dart';

import '../../../room.dart';

abstract interface class BrowseRoomRepository {
  Future<Either<Failure, List<AvailableRoomEntity>>> getAvailableRooms({
    double? minRent,
    double? maxRent,
  });

  Future<Either<Failure, BrowseRoomDetailEntity>> getRoomDetail({
    required String id,
  });
}

File: server/src/modules/rooms/domain/repositories/browse-room.repository.ts
import { AvailableRoomEntity, BrowseRoomDetailEntity } from '../entities/browse-room.entity';

export type BrowseRoomFilters = {
  minRent?: number;
  maxRent?: number;
};

export abstract class BrowseRoomRepository {
  abstract findAvailable(filters: BrowseRoomFilters): Promise<AvailableRoomEntity[]>;
  abstract findDetailById(id: string): Promise<BrowseRoomDetailEntity | null>;
}
